### DISCUSSION TITLE
Patch and Comms headers

### DISCUSSION THREAD

**Discussion by donfroula** [thehighnibble/thehighnibble.github.io#168] [IMSAI8080esp replica]
I'm curious about the "Patch" and "Comms" jumpers functions.

The functions and labeling of the Patch header pins is puzzling.

The Patch headers seem to jumper to physical UART pins on the ESP32, or specific pins associated with emulated Z80 ports.

I think that the Comms headers connect TTL-level RS232 signals from the Patch header to the level converter chip and thus to the DB9 connectors.

I noticed Comms jumpers for RTS/CTS hardware handshaking, but the recommended setup does not have these pins connected. Do these jumpers connect to emulated HAL UART functionality, or to physical pins on the ESP32? Is hardware flow control even possible?

**Comment by morphinejh** [comment]
I am also interested if hardware flow control is possible - is it implemented in firmware? It looks like if you disabled one of the RS-232 ports, you would have gain enough pins to implement it with careful wiring of the pin headers. But the firmware would have to support first.
